src/pages/setup.js

```diff
@@ -13,10 +13,14 @@ class SetupPage extends Component {
     const { router } = this.props;
     const name = InstanceDialogStore.genUniqueName();
 
-    connection.Instance.please().create({ name }).then((instance) => {
-      SessionActions.fetchInstance(instance.name);
-      router.push(`/instances/${instance.name}/`);
-    });
+    if (SessionStore.getSignUpMode()) {
+      connection.Instance.please().create({ name }).then((instance) => {
+        SessionActions.fetchInstance(instance.name);
+        router.push(`/instances/${instance.name}/`);
+      });
+
+      SessionStore.removeSignUpMode();
```

Comment:
it should be done in the callback after `router.push` to make sure it fires after a first Instance has been created